File: configuration/src/test/java/com/synopsys/integration/configuration/property/types/longs/LongValueParserTests.java
package com.synopsys.integration.configuration.property.types.longs;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;

import com.synopsys.integration.configuration.parse.ValueParseException;

public class LongValueParserTests {
    private final LongValueParser parser = new LongValueParser();

    @ParameterizedTest
    @ValueSource(strings = { "unknown", "Nan", "", " 1", "1L", "9223372036854775808" })
    public void parseUnknownThrows(String value) {
        Assertions.assertThrows(ValueParseException.class, () -> parser.parse(value));
    }

    @Test
    public void parseLong() throws ValueParseException {
        Assertions.assertEquals(new Long(-1), parser.parse("-1"));
        Assertions.assertEquals(new Long(1), parser.parse("1"));
        Assertions.assertEquals(new Long(Long.MAX_VALUE), parser.parse("9223372036854775807"));
        Assertions.assertEquals(new Long(Long.MIN_VALUE), parser.parse("-9223372036854775808"));
    }
}

File: configuration/src/test/java/com/synopsys/integration/configuration/config/long/LongValueParserTests.kt
package com.synopsys.integration.configuration.config.long

import com.synopsys.integration.configuration.parse.ValueParseException
import com.synopsys.integration.configuration.property.types.longs.LongValueParser
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.ValueSource

class LongValueParserTests {
    @ParameterizedTest
    @ValueSource(strings = ["unknown", "Nan", "", " 1", "1L", "9223372036854775808"])
    fun parseUnknownThrows(value: String) {
        Assertions.assertThrows(ValueParseException::class.java) {
            LongValueParser().parse(value)
        }
    }

    @Test
    fun parseLong() {
        fun assert(expected: Long, value: String) {
            Assertions.assertEquals(expected, LongValueParser().parse(value))
        }

        assert(-1, "-1")
        assert(1, "1")
        assert(Long.MAX_VALUE, "9223372036854775807")
        assert(Long.MIN_VALUE, "-9223372036854775808")
    }
}